### PR TITLE
fix #217: correct input$sidebarCollapsed value for edge cases

### LINKED ISSUE #217: Sidebar collapsed/expanded is reversed for small width/mobile
This script:

```r
library(shiny)
library(shinydashboard)

ui <- dashboardPage(
  dashboardHeader(), 
  dashboardSidebar(),
  dashboardBody(
    textOutput("res")
  )
)

server <- function(input, output, session) {
  output$res <- renderText({
    if (input$sidebarCollapsed) {
      "Sidebar is collapsed"
    } else {
      "Sidebar is expanded"
    }
  })
}

shinyApp(ui, server)
```

works as expected in a laptop-sized screen:
screen shot 2017-05-21 at 9 40 22 pm
screen shot 2017-05-21 at 9 40 29 pm

but it's reversed for small-width/mobile screens:

image 1
image

### Note
This is a smaller problem than it appears because I suspect no-one (or very few people) are actually interested in the value of `input$sidebarCollapsed`. Rather, they just want the sidebar to look the same once they bookmark their app (which will happen as long as `input$sidebarCollapsed` has the same value, even if it is the wrong value). Nonetheless, this should be fixed.

### PR DESCRIPTION
**Pull request by bborgesr**
Attach input$sidebarCollapsed value to the end of the sidebar transitions (instead of when the toggle button is clicked).

Repro:
```r
library(shiny)
library(shinydashboard)

ui <- dashboardPage(
  dashboardHeader(), 
  dashboardSidebar(),
  dashboardBody(
    textOutput("res")
  )
)

server <- function(input, output, session) {
  output$res <- renderText({
    if (input$sidebarCollapsed) {
      "Sidebar is collapsed"
    } else {
      "Sidebar is expanded"
    }
  })
}

shinyApp(ui, server)
```

### CODE REVIEW

**Comment by bborgesr** [comment]
@wch This is rebased, updated and tested as much as I could think of... I think we're good to merge.

File: srcjs/sidebar.js [hunk 1]
```diff
@@ -5,17 +5,46 @@ if ($("section.sidebar").data("disable")) {
...
+    sidebarChange);
+
+// This fixes an edge case: when the app starts up with the sidebar already
+// collapsed (either because the screen is small or because
+// `dashboardSidebar(disable = TRUE`), make sure that `input$sidebarCollapsed`
+// is set to `FALSE`. Whenever this is the case, `$(".main-sidebar").offset().left`
+// is negative. That we know of, `$(".main-sidebar").is(":visible")` is always
+// true, so there is no need to check for that.
+if ($(".main-sidebar").offset().left < 0) {
+  $(".main-sidebar").attr("data-collapsed", "true");
```
Comment (wch) [inline srcjs/sidebar.js]: This is a little hard to follow since it sets the `data-collapsed` attribute directly, whereas in the `sidebarChange` function, it uses `inputBinding.setValue()`. Looking at the code, it's not obvious that they are both controlling the same thing.

Would it be possible to use the input binding here, or is this code executed too early for that to work?

Comment (bborgesr) [inline srcjs/sidebar.js]: That's right. I tried the first approach (`inputBinding.setValue()`), but it's too early to call here. I managed to get it working by listening on the first shiny idle event (or something along those lines), but not only is that adding extra complexity unnecessarily, but also, and worse, the visual effect is weird if you're actually accessing that value. For example, in this app:

```r
library(shiny)
library(shinydashboard)

ui <- dashboardPage(
  dashboardHeader(), 
  dashboardSidebar(),
  dashboardBody(
    textOutput("res")
  )
)

server <- function(input, output, session) {
  output$res <- renderText({
    if (input$sidebarCollapsed) {
      "Sidebar is collapsed"
    } else {
      "Sidebar is expanded"
    }
  })
}

shinyApp(ui, server)
```

, you'd see the string starting off as "Sidebar is expanded" and then immediately changing to "Sidebar is collapsed."

I felt like this was okay, because (aside from the fact that shinydashboard is already full of hacks for the input bindings), we also do the exact same thing [here](https://github.com/rstudio/shinydashboard/blob/master/srcjs/tabs.js#L47-L48) for the selected tab item... What do you think? I could add another comment in both sections about why we have to go about this in this way.
